src/main/kotlin/net/atos/zac/zoeken/model/zoekobject/DocumentZoekObject.kt

```diff
@@ -0,0 +1,171 @@
+
+    @Field("informatieobject_ontvangstdatum")
+    var ontvangstdatum: Date? = null,
+
+    @Field("informatieobject_verzenddatum")
+    var verzenddatum: Date? = null,
+
+    @Field("informatieobject_ondertekeningDatum")
+    var ondertekeningDatum: Date? = null,
+
+    @Field("informatieobject_ondertekeningSoort")
+    var ondertekeningSoort: String? = null,
+
+    @Field("informatieobject_vertrouwelijkheidaanduiding")
+    var vertrouwelijkheidaanduiding: String? = null,
+
+    @Field("informatieobject_auteur")
+    var auteur: String? = null,
+
+    @Field("informatieobject_status")
+    private var status: String? = null,
+
+    @Field("informatieobject_formaat")
+    var formaat: String? = null,
+
+    @Field("informatieobject_versie")
+    var versie: Long = 0,
+
+    @Field("informatieobject_bestandsnaam")
+    var bestandsnaam: String? = null,
+
+    @Field("informatieobject_bestandsomvang")
+    var bestandsomvang: Long = 0,
+
+    @Field("informatieobject_documentType")
+    var documentType: String? = null,
+
+    @Field("informatieobject_inhoudUrl")
+    var inhoudUrl: String? = null,
+
+    @Field("informatieobject_vergrendeldDoorNaam")
+    var vergrendeldDoorNaam: String? = null,
+
+    @Field("informatieobject_vergrendeldDoorGebruikersnaam")
+    var vergrendeldDoorGebruikersnaam: String? = null,
+
+    @Field("informatieobject_indicaties")
+    private var indicaties: MutableList<String>? = null,
+
+    @Field("informatieobject_indicaties_sort")
+    private var indicatiesVolgorde: Long = 0
+) : ZoekObject {
+    override fun getObjectId() = uuid
+
+    override fun getType() = ZoekObjectType.valueOf(type)
+
+    fun setType(type: ZoekObjectType) {
+        this.type = type.toString()
+    }
+
+    fun getStatus() = status?.uppercase(Locale.getDefault())?.let { StatusEnum.valueOf(it) }
+
+    fun setStatus(status: StatusEnum) {
+        this.status = status.toString()
+    }
+
+    fun isIndicatie(indicatie: DocumentIndicatie) = indicaties?.contains(indicatie.name) == true
+
+    fun getDocumentIndicaties(): EnumSet<DocumentIndicatie> =
+        if (indicaties == null) {
+            EnumSet.noneOf<DocumentIndicatie>(DocumentIndicatie::class.java)
+        } else {
+            indicaties!!.stream().map<DocumentIndicatie>(DocumentIndicatie::valueOf)
+                .collect(
+                    Collectors.toCollection(
+                        Supplier {
+                            EnumSet.noneOf<DocumentIndicatie>(DocumentIndicatie::class.java)
+                        }
+                    )
+                )
+        }
+
+    fun setIndicatie(indicatie: DocumentIndicatie, value: Boolean) {
+        updateIndicaties(indicatie, value)
+        updateIndicatieVolgorde(indicatie, value)
+    }
+
+    private fun updateIndicaties(indicatie: DocumentIndicatie, value: Boolean) {
+        val key = indicatie.name
+        if (indicaties == null) {
+            indicaties = arrayListOf()
+        }
+        indicaties?.let {
```

Comment:
I don't think this works properly as you'd like... and if it has just been initialized above, then there's no need handle nullability here, and probably not even bother with the let, just do the checks directly on the `indicaties` instead of `it`.

Reply:
yeah true